# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentLoadListener.cpp
# Commit: acafbde0a546
# Full Hash: acafbde0a54602fcd1a634ddd772d666f7e96455
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2021-03-09 16:11:38
# Regressor Bug: 1696266
# File Overlap Count: 2
# Description:
#   Bug 1696266, limit the load types which may cause the page to enter bfcache, r=peterv
#   
#   The change to dom/base/nsFrameLoaderOwner.cpp is to log about the issues but still ensure we don't crash.
#   
#   I'd prefer to not put error loads to bfcache.
# ==============================================================================

diff -r f0d286830ebb -r acafbde0a546 netwerk/ipc/DocumentLoadListener.cpp
--- a/netwerk/ipc/DocumentLoadListener.cpp	Tue Mar 09 13:34:51 2021 +0000
+++ b/netwerk/ipc/DocumentLoadListener.cpp	Tue Mar 09 13:42:43 2021 +0000
@@ -1631,7 +1631,10 @@
       browsingContext->Group()->Toplevels().Length() == 1 &&
       !options.mRemoteType.IsEmpty() &&
       browsingContext->GetHasLoadedNonInitialDocument() &&
-      mLoadStateLoadType != LOAD_ERROR_PAGE) {
+      (mLoadStateLoadType == LOAD_NORMAL ||
+       mLoadStateLoadType == LOAD_HISTORY || mLoadStateLoadType == LOAD_LINK ||
+       mLoadStateLoadType == LOAD_STOP_CONTENT ||
+       mLoadStateLoadType == LOAD_STOP_CONTENT_AND_REPLACE)) {
     options.mReplaceBrowsingContext = true;
     options.mTryUseBFCache = true;
   }
